document-store/src/main/java/org/hypertrace/core/documentstore/Collection.java

```diff
@@ -80,6 +80,15 @@ public interface Collection {
    */
   CloseableIterator<Document> aggregate(final org.hypertrace.core.documentstore.query.Query query);
 
+  /**
+   * Count the result-set size of applying aggregations. Note that this method is a generic version
+   * of {@link #total(Query)}
+   *
+   * @param query The query definition whose result-set size is to be determined
+   * @return The number of documents conforming to the input query
+   */
+  long aggregateCount(final org.hypertrace.core.documentstore.query.Query query);
```

Comment:
If the method is a generic version of `@link #total(Query)`, shouldn't we name it more generically?

Reply:
Right. Renamed it.